src/handlers/pageRouter/callback.js

```diff
@@ -1,18 +1,34 @@
+import jwt_decode from 'jwt-decode';
+
 import {config} from '../../config/index';
 import {isTokenValid} from '../../utils/pageRouter/isTokenValid';
 import {version} from '../../utils/version';
-import jwt_decode from 'jwt-decode';
+import {sanitizeRedirect} from '../../utils/sanitizeRedirect';
 
 var cookie = require('cookie');
 
 export const callback = async (req, res) => {
   const {code, state} = req.query;
-  const code_verifier = cookie.parse(req.headers.cookie || '')[
+  const jsonCookieValue = cookie.parse(req.headers.cookie || '')[
     `${config.SESSION_PREFIX}-${state}`
   ];
 
-  if (code_verifier) {
+  let redirectUrl = config.postLoginURL || config.redirectURL;
+
+  if (jsonCookieValue) {
     try {
+      const {
+        code_verifier,
+        options,
+      } = JSON.parse(jsonCookieValue);
+
+      if (options?.callback_url) {
+        redirectUrl = sanitizeRedirect({
+          baseUrl: new URL(config.redirectURL).origin,
+          url: options.callback_url
```

Comment:
@kengreeff all looks awesome. The only thing I'd say it I'm not sure on calling the option `callback_url` just because it has a specific meaning in oauth flows that it is the url being called back to after authentication to complete the flow. Whereas this is specifically the url to pass onto post-authenticating and after the callback has completed.

Maybe `next_url`? 

What are your thoughts @peterphanouvong ?

Reply:
How about `redirect_url`?